schism/mplink.c

```diff
@@ -222,6 +222,29 @@ int song_find_last_channel(void)
 
 // ------------------------------------------------------------------------
 
+int song_get_pattern_offset(int * n, song_note_t ** buf, int * r, int offset)
+{
+	int len, tot;
+	if (song_get_mode() & (MODE_PATTERN_LOOP)) {
+		// just wrap around current rows
+		*r = (*r + offset) % song_get_rows_in_pattern(*n);
+		return song_get_pattern(*n, buf);
```

Comment:
This doesn't change the contents of `buf`, right, since we're not changing patterns? In that case you could pass `NULL` instead of `buf` here.

Reply:
what do you mean by
> since we're not changing patterns

this function replaces the original `song_get_pattern` call in `pattern_editor_insert_midi` to select and instantiate the pattern buffer with the right #n pattern according to the offset.

`buf` is still NULL here for that matter

Reply:
Never mind, I disagree with what I said originally.